src/equations/shallow_water_multilayer_2d.jl

```diff
@@ -0,0 +1,827 @@
+        issorted(rhos) ||
+            throw(ArgumentError("densities must be in increasing order (rhos[1] < rhos[2] < ... < rhos[NLAYERS])"))
+        min(rhos...) > 0 || throw(ArgumentError("densities must be positive"))
+
+        new(gravity, H0, rhos)
+    end
+end
+
+# Allow for flexibility to set the gravitational constant within an elixir depending on the
+# application where `gravity_constant=1.0` or `gravity_constant=9.81` are common values.
+# The reference total water height H0 defaults to 0.0 but is used for the "lake-at-rest"
+# well-balancedness test cases. 
+function ShallowWaterMultiLayerEquations2D(; gravity_constant,
+                                           H0 = zero(gravity_constant), rhos)
+
+    # Promote all variables to a common type
+    _rhos = promote(rhos...)
+    RealT = promote_type(eltype(_rhos), eltype(gravity_constant), eltype(H0))
+    __rhos = SVector(map(RealT, _rhos))
+
+    # Extract number of layers and variables
+    NLAYERS = length(rhos)
+    NVARS = 3 * NLAYERS + 1
+
+    return ShallowWaterMultiLayerEquations2D{NVARS, NLAYERS, RealT}(gravity_constant,
+                                                                    H0, __rhos)
+end
+
+@inline function Base.real(::ShallowWaterMultiLayerEquations2D{NVARS, NLAYERS, RealT}) where {
+                                                                                              NVARS,
+                                                                                              NLAYERS,
+                                                                                              RealT <:
+                                                                                              Real
+                                                                                              }
+    RealT
+end
+
+Trixi.have_nonconservative_terms(::ShallowWaterMultiLayerEquations2D) = True()
+function Trixi.varnames(::typeof(cons2cons),
+                        equations::ShallowWaterMultiLayerEquations2D)
+    waterheight = ntuple(n -> "h" * string(n), Val(nlayers(equations)))
+    momentum_1 = ntuple(n -> "h" * string(n) * "_v1", Val(nlayers(equations)))
+    momentum_2 = ntuple(n -> "h" * string(n) * "_v2", Val(nlayers(equations)))
+    return (waterheight..., momentum_1..., momentum_2..., "b")
+end
+
+# We use the total layer heights, H = ∑h + b as primitive variables for easier visualization and setting initial
+# conditions
+function Trixi.varnames(::typeof(cons2prim),
+                        equations::ShallowWaterMultiLayerEquations2D)
+    total_layer_height = ntuple(n -> "H" * string(n), Val(nlayers(equations)))
+    velocity_1 = ntuple(n -> "v" * string(n) * "_1", Val(nlayers(equations)))
+    velocity_2 = ntuple(n -> "v" * string(n) * "_2", Val(nlayers(equations)))
+    return (total_layer_height..., velocity_1..., velocity_2..., "b")
+end
+
+# # Set initial conditions at physical location `x` for time `t`
```

Comment:
```suggestion
# Set initial conditions at physical location `x` for time `t`
```

Reply:
Done in https://github.com/trixi-framework/TrixiShallowWater.jl/pull/40/commits/2eec19df7e184318796af5c0c14f2be8673918ab